doc/mgr/smb.rst

```diff
@@ -952,3 +952,61 @@ at the prompt. Refer to the `smbclient documentation`_ for more details.
 
 .. _smbclient documentation:
    https://www.samba.org/samba/docs/current/man-html/smbclient.1.html
+
+
+Updating QoS Settings for CephFS-Backed Shares
+=================================================
+
+The SMB manager module now supports updating *QoS (Quality of Service)* settings
+for CephFS-backed shares. This allows administrators to apply per-share rate
+limits on SMB I/O operations, specifically limits on the number of operations
+per second (IOPS) and bandwidth in bytes per second, for both read and
+write operations.
+
+QoS Update Command:
+.. code-block:: console
+
+    ceph smb share update cephfs qos <cluster_id> <share_id> \
+      [--read-iops-limit=<int>] \
+      [--write-iops-limit=<int>] \
+      [--read-bw-limit=<int>] \
+      [--write-bw-limit=<int>]
+
+Behavior:
+- Each parameter is optional
+- Setting a limit to ``0`` disables that specific QoS limit
+- Setting all four limits to ``0`` completely removes QoS configuration
+
+Example: Setting Limits:
+.. code-block:: console
+
+    ceph smb share update cephfs qos foo bar \
+      --read-iops-limit=100 \
+      --write-iops-limit=200 \
+      --read-bw-limit=1048576 \
+      --write-bw-limit=2097152
+
+To completely disable QoS for a share:
+.. code-block:: console
+
+    ceph smb share update cephfs qos foo bar \
+      --read-iops-limit=0 \
+      --write-iops-limit=0 \
+      --read-bw-limit=0 \
+      --write-bw-limit=0
+
+Parameter Details:
+- ``read_iops_limit`` / ``write_iops_limit``:
+  Maximum number of SMB read/write operations per second (0 = disabled)
+- ``read_bw_limit`` / ``write_bw_limit``:
+  Maximum allowed bandwidth for SMB read/write in bytes per second (0 = disabled)
+
+Resulting configuration:
+.. code-block:: yaml
+
+    cephfs:
+      qos:
+        read_iops_limit: 100
+        write_iops_limit: 200
+        read_bw_limit: 1048576
+        write_bw_limit: 2097152
```

Comment:
If I didn't know the structure of our resources in YAML extremely well, I think I might find this confusing. I suggest filling out more of the yaml OR at the very least make sure it's clear this is a snippet of YAML and not a complete resource spec.

Reply:
Done